### DISCUSSION TITLE
`noUnsafeReturn`, based on typescript-eslint/no-unsafe-return

### DISCUSSION THREAD

**Discussion by jpike88** [biomejs/biome#340] [Ideas]
I ran into a nasty bit of code smell today, where an any was bubbled up through some functions to be returned by a function with a different signature.

This rule would have prevented that, and ensured that the 'any' variable was handled or coerced into a type before being returned anywhere.

https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-unsafe-return.md

**Comment by jpike88** [comment]
example code that would have been allowed if this rule wasn't on...

```
function wtf():Promise<number>{
  const a: any = 'wtf';
  return a;
}
```

**Comment by Conaclos** [comment]
Thanks to bring it to light! I added this rule in the [list of rules we want to support](https://github.com/biomejs/biome/discussions/3).
